bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -151,32 +147,26 @@ private void getPrices() {
             String content = contentResponse.getContentAsString();
             logger.trace("aWATTar API response: status = {}, content = '{}'", httpStatus, content);
 
-            switch (httpStatus) {
-                case OK_200:
-                    Gson gson = new Gson();
-                    SortedMap<Long, AwattarPrice> result = new TreeMap<>();
-                    minTimestamp = 0;
-                    maxTimestamp = 0;
-                    AwattarApiData apiData = gson.fromJson(content, AwattarApiData.class);
-                    if (apiData != null) {
-                        for (Datum d : apiData.data) {
-                            result.put(d.startTimestamp,
-                                    new AwattarPrice(d.marketprice / 10.0, d.startTimestamp, d.endTimestamp, zone));
-                            updateMin(d.startTimestamp);
-                            updateMax(d.endTimestamp);
-                        }
-                        priceMap = result;
-                        updateStatus(ThingStatus.ONLINE);
-                        lastUpdated = Instant.now().toEpochMilli();
-                    } else {
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                                "@text/error.invalid.data");
+            if (httpStatus == OK_200) {
+                Gson gson = new Gson();
+                SortedSet<AwattarPrice> result = new TreeSet<>(Comparator.comparing(AwattarPrice::timerange));
```

Comment:
@Wolfgang1966 See here: a `Comparator` for the time ranges is used to sort the set.